File: CSharp/FizzBuzz.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FizzBuzzCS
{
    class FizzBuzz
    {
        static void Main(string[] args)
        {
            // Multiples and fizzbuzz strings
            string[] fizzBuzz = { "Fizz", "Buzz" };
            int[] multiples = { 3, 5 };

            // Loop from 1 to 100
            for (int i = 1; i <= 100; i++)
            {
                // Empty string 
                string output = "";
                for (int j = 0; j < multiples.Length; j++)
                {
                    // If a multiple
                    if (i % multiples[j] == 0)
                    {
                        // Add the relevant string
                        output += fizzBuzz[j];
                    }
                }
                // If the string is empty it is not a multiple of anything
                if (output == "")
                {
                    // Print the number
                    Console.WriteLine(i);
                }
                else
                {
                    Console.WriteLine(output);
                }
            }
        }
    }
}

File: cpp/Fizzbuzz.cpp
#include <iostream>
#include <string>

int main() {
	std::string fizzBuzz[] = { "Fizz", "Buzz" };
	int multiples[] = { 3, 5 };

	//from 1 to 100
	for (int i = 1; i <= 100; i++) {
		std::string output = "";
		//for each multiple
		for (int j = 0; j <= (sizeof(multiples) / sizeof(*multiples)) - 1; j++) {
			//if a multiple of j
			if (i % multiples[j] == 0) {
				output = output.append(fizzBuzz[j]);
			}
		}
		//if not a multiple of the given numbers
		if (output == "") {
			std::cout << i << "\n";
		}
		else {
			std::cout << output << "\n";
		}
	}
}